sktime/forecasting/hf_transformers_forecaster.py

```diff
@@ -0,0 +1,398 @@
+        for details.
+    deterministic : bool, default=False
+        Whether the predictions should be deterministic or not.
+    callbacks : list, default=[]
+        List of callbacks to use during training. See `transformers.Trainer`
+
+    Examples
+    --------
+    >>> from sktime.forecasting.hf_transformers_forecaster import (
+    ...     HFTransformersForecaster,
+    ... )
+    >>> from sktime.datasets import load_airline
+    >>> y = load_airline()
+    >>> forecaster = HFTransformersForecaster(
+    ...    model_path="huggingface/autoformer-tourism-monthly",
+    ...    training_args ={
+    ...        "num_train_epochs": 20,
+    ...        "output_dir": "test_output",
+    ...        "per_device_train_batch_size": 32,
+    ...    },
+    ...    config={
+    ...         "lags_sequence": [1, 2, 3],
+    ...         "context_length": 2,
+    ...         "prediction_length": 4,
+    ...         "use_cpu": True,
+    ...         "label_length": 2,
+    ...    },
+    ... ) # doctest: +SKIP
+    >>> forecaster.fit(y) # doctest: +SKIP
+    >>> fh = [1, 2, 3]
+    >>> y_pred = forecaster.predict(fh) # doctest: +SKIP
+    """
+
+    _tags = {
+        "ignores-exogeneous-X": False,
+        "requires-fh-in-fit": False,
+        "X-y-must-have-same-index": True,
+        "enforce_index_type": None,
+        "handles-missing-data": False,
+        "capability:pred_int": False,
+        "python_dependencies": ["transformers", "torch"],
+        "X_inner_mtype": "pd.DataFrame",
+        "y_inner_mtype": "pd.Series",
+        "capability:insample": False,
+        "capability:pred_int:insample": False,
+    }
+
+    def __init__(
+        self,
+        model_path: str,
+        fit_strategy="minimal",
+        validation_split=0.2,
+        config=None,
+        training_args=None,
+        compute_metrics=None,
+        deterministic=False,
+        callbacks=None,
+    ):
+        super().__init__()
+        self.model_path = model_path
+        self.fit_strategy = fit_strategy
+        self.validation_split = validation_split
+        self.config = config
+        self._config = config if config is not None else {}
+        self.training_args = training_args
+        self._training_args = training_args if training_args is not None else {}
+        self.compute_metrics = compute_metrics
+        self._compute_metrics = compute_metrics
+        self._compute_metrics = compute_metrics
+        self.deterministic = deterministic
+        self.callbacks = callbacks
+        self._callbacks = callbacks
+
+    def _fit(self, y, X, fh):
+        # Load model and extract config
+        config = AutoConfig.from_pretrained(self.model_path)
+
+        # Update config with user provided config
+        _config = config.to_dict()
+        _config.update(self._config)
+        _config["num_dynamic_real_features"] = X.shape[-1] if X is not None else 0
+        _config["num_static_real_features"] = 0
+        _config["num_dynamic_real_features"] = 0
```

Comment:
Not a review, just a question: what is th difference between `num_dynamic_real_features` and `num_time_features`?

Reply:
To be honest. I am not quite sure:

This is the documentation from hugging face: https://huggingface.co/docs/transformers/model_doc/autoformer#transformers.AutoformerConfig

```

num_time_features (int, optional, defaults to 0) — The number of time features in the input time series.

num_dynamic_real_features (int, optional, defaults to 0) — The number of dynamic real valued features.

num_static_categorical_features (int, optional, defaults to 0) — The number of static categorical features.
```

Based on that I decided to just ignore num_dynamic_real_features and put everything into num_time_features